packages/api/src/middleware/farm.js

```diff
@@ -0,0 +1,143 @@
+
+export function parseMultipartJson(req, res, next) {
+  const contentType = req.get('Content-Type');
+
+  if (contentType.includes('multipart/form-data') && req.body?.data) {
+    try {
+      req.body = JSON.parse(req.body.data);
+    } catch (error) {
+      return res.status(400).json({ error: error.message });
+    }
+  }
+
+  next();
+}
+
+// handles uploading, deleting, and replacing farm image
+export async function handleImageOperations(req, res, next) {
+  if (!req.file && !req.body.shouldRemoveImage) return next();
+
+  const farmId = req.params.farm_id;
+
+  try {
+    const existingImageKeys = await getExistingImageKeys(farmId);
+    const privateS3Url = getPrivateS3Url();
+
+    const setFarmImageUrls = (keys) => {
+      req.body.farm_image_url = keys ? `${privateS3Url}/${keys.imageKey}` : null;
+      req.body.farm_image_thumbnail_url = keys ? `${privateS3Url}/${keys.thumbnailKey}` : null;
+    };
+
+    if (req.file) {
+      const newImageKeys = {
+        imageKey: `${farmId}/farm_image/${getRandomFileName(req.file)}`,
+        thumbnailKey: `${farmId}/farm_image_thumbnail/${uuidv4()}.webp`,
+      };
+
+      if (existingImageKeys) {
+        await deleteFarmImage(existingImageKeys);
+        await uploadFarmImage(req.file, newImageKeys);
+      } else {
+        await uploadFarmImage(req.file, newImageKeys);
+      }
+
+      setFarmImageUrls(newImageKeys);
+      return next();
+    }
+
+    if (req.body.shouldRemoveImage && existingImageKeys) {
+      await deleteFarmImage(existingImageKeys);
+      setFarmImageUrls();
+    }
+  } catch (error) {
+    console.error(error);
+    return res.sendStatus(500);
+  }
+
+  next();
+}
+
+async function uploadFarmImage(imageFile, keys) {
+  const bucketName = getPrivateS3BucketName();
+  const thumbnail = await imaginaryPost(imageFile, {
+    width: 150,
+    type: 'webp',
+  });
+
+  const thumbnailUploadPromise = s3
+    .putObject({
+      Body: thumbnail.data,
+      Bucket: bucketName,
+      Key: keys.thumbnailKey,
+      ACL: 'private',
+    })
+    .promise();
+
+  const imageUploadPromise = s3
+    .putObject({
+      Body: imageFile.buffer,
+      Bucket: bucketName,
+      Key: keys.imageKey,
+      ACL: 'private',
+    })
+    .promise();
+
+  await Promise.all([thumbnailUploadPromise, imageUploadPromise]);
+}
+
+async function deleteFarmImage({ thumbnailKey, imageKey }) {
+  const bucketName = getPrivateS3BucketName();
+
+  const deleteImagePromsie = s3.deleteObject({ Bucket: bucketName, Key: imageKey }).promise();
+
+  const deleteThumbnailPromise = s3
+    .deleteObject({ Bucket: bucketName, Key: thumbnailKey })
+    .promise();
+
+  await Promise.all([deleteImagePromsie, deleteThumbnailPromise]);
+}
+
+export async function getExistingImageKeys(farmId) {
+  const {
+    farm_image_url: imageUrl,
+    farm_image_thumbnail_url: thumbnailUrl,
+  } = await FarmModel.query().findById(farmId);
+
+  if (!imageUrl && !thumbnailUrl) return;
```

Comment:
This should probably be

```suggestion
  if (!imageUrl || !thumbnailUrl) return;
```

If either url for some weird reason is not defined, it will produce an error in the function below when attempting to parse that string